api/include/opentelemetry/trace/semantic_conventions.h

```diff
@@ -0,0 +1,190 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/version.h"
+
+#define OTEL_TRACE_GENERATE_SEMCONV_METHOD(attribute_name, attribute_value) \
+  static const nostd::string_view Get##attribute_name() noexcept            \
+  {                                                                         \
+    static const nostd::string_view attribute_name = attribute_value;       \
+    return attribute_name;                                                  \
+  }
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace trace
+{
+
+/**
+ * Stores the Constants for semantic kAttribute names outlined by the OpenTelemetry specifications.
+ * <see
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/README.md"/>.
+ */
+class SemanticConventions final
+{
+public:
+  // The set of constants matches the specification as of this commit.
+  // https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md
+
+  // Genetal network connection attributes
+  OTEL_TRACE_GENERATE_SEMCONV_METHOD(AttributeNetTransport, "net.transport")
```

Comment:
Expand `Attribute` in name `AttributeNetTransport` in the macro of `OTEL_TRACE_GENERATE_SEMCONV_METHOD` instead of duplicating it for every macro call?

Reply:
Good point. Will do  that.

Reply:
This is done now.